src/platform/backends/shared_linux/backend_utils.cpp

```diff
@@ -130,23 +134,48 @@ void mp::backend::check_hypervisor_support()
     }
 }
 
-void mp::backend::resize_instance_image(const std::string& disk_space, const mp::Path& image_path)
+void mp::backend::resize_instance_image(const ProcessFactory* process_factory, const std::string& disk_space,
+                                        const mp::Path& image_path)
 {
     auto disk_size = QString::fromStdString(disk_space);
 
     if (disk_size.endsWith("B"))
         disk_size.chop(1);
 
-    if (!mp::utils::run_cmd_for_status("qemu-img", {QStringLiteral("resize"), image_path, disk_size}))
+    auto qemuimg_spec = std::make_unique<mp::QemuImgProcessSpec>();
+    auto qemuimg_process = process_factory->create_process(std::move(qemuimg_spec));
+
+    if (!qemuimg_process->run_and_return_status({"resize", image_path, disk_size}))
         throw std::runtime_error("Cannot resize instance image");
 }
 
-std::string mp::backend::image_format_for(const mp::Path& image_path)
+mp::Path mp::backend::convert_to_qcow_if_necessary(const ProcessFactory* process_factory, const mp::Path& image_path)
 {
-    auto image_info = QString::fromStdString(
-        mp::utils::run_cmd_for_output("qemu-img", {QStringLiteral("info"), "--output=json", image_path}));
+    // Check if raw image file, and if so, convert to qcow2 format.
```

Comment:
Add a TODO that we could convert from other image formats as necessary?

Reply:
Done